test/shmemx/shmemx_wait_until_some.c

```diff
@@ -0,0 +1,50 @@
+#include <shmem.h>
+#include <shmemx.h>
+#include <stdlib.h>
+
+#define N 100
+
+int main(void)
+{
+  int total_sum = 0;
+
+  shmem_init();
+  int mype = shmem_my_pe();
+  int npes = shmem_n_pes();
+
+  int *my_data = malloc(N * sizeof(int));
+  int *all_data = shmem_malloc(N * npes * sizeof(int));
+
+  int *flags = shmem_calloc(npes, sizeof(int));
+  size_t *indices = malloc(npes * sizeof(size_t));
+  int *status = calloc(npes, sizeof(int));
+
+  for (int i = 0; i < N; i++)
+      my_data[i] = mype*N + i;
+
+  for (int i = 0; i < npes; i++)
+      shmem_int_put_nbi(&all_data[mype*N], my_data, N, i);
+
+  shmem_fence();
+
+  for (int i = 0; i < npes; i++)
+      shmem_int_p(&flags[mype], 1, i);
+  
+  size_t ncompleted;
+  while ((ncompleted = shmemx_int_wait_until_some(flags, npes, indices,
+                                                  status, SHMEM_CMP_NE, 0))) {
+      for (size_t i = 0; i < ncompleted; i++) {
+          for (size_t j = 0; j < N; j++)
+              total_sum += all_data[indices[i]*N + j];
+      }
+  }
+
+  /* check result */
+  int M = N * npes - 1;
+  if (total_sum != M * (M + 1) / 2) {
+      shmem_global_exit(1);
+  }
+
+  shmem_finalize();
+  return 0;
+}
```

Comment:
Coverage-wise, it might be good to work in a test that passes NULL for status.

Reply:
Whoops!  Sorry, the whitespace changes here might be misleading.  Everything is the same except for the "sanity check" for the `status == NULL` case.